hostpool_test.go

```diff
@@ -174,3 +174,39 @@ func BenchmarkEpsilonGreedyManyHosts(b *testing.B) {
 		hostR.Mark(nil)
 	}
 }
+
+func TestHostPoolErrorBudget(t *testing.T) {
+	log.SetOutput(ioutil.Discard)
+	defer log.SetOutput(os.Stdout)
+
+	dummyErr := errors.New("Dummy Error")
+
+	p := NewWithOptions([]string{"a", "b"}, StandardHostPoolOptions{
+		maxFailures: 2,
```

Comment:
this test works because we're relying on the default failure window (60s?) always being greater than the time to execute the test, right? maybe we should include an explicit value in the setup so it's more obvious.

perhaps we could also add a test with a low failure window + time.Sleep to demonstrate the reset behaviour?

Reply:
Yes that's right, I'll add an explicit value 👍